### PR TITLE
Fixed the MT

### PR DESCRIPTION
**Pull request by rAguilarJala**
I send you PR with fixes, please check it.

### CODE REVIEW

File: IDT.MoneyTransfer/Main/UI/Entities/TransferDetailsEntity.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Entities/TransferDetailsEntity.cs]: Update the class con Code standards

File: IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 {
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs]: Add Header documentation

File: IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs [hunk 3]
```diff
@@ -2,6 +2,7 @@
...
     using IDT.Core.Automation.Page.Base;
     using IDT.Core.Automation.Selenium;
+    using IDT.MoneyTransfer.Main.Utilities;
     using OpenQA.Selenium;
 
     class HomePage : BasePage
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs [hunk 4]
```diff
@@ -14,6 +15,7 @@ class HomePage : BasePage
 
         public ProductsTab PressTabProduct()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs [hunk 5]
```diff
@@ -14,6 +15,7 @@ class HomePage : BasePage
...
         {
+            CommonActions.WaitUntil(2);
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs]: Why we are using this wait ? Is it not possible to use WaitUntilElementIsDisplayed(IWebElement webElement)??

File: IDT.MoneyTransfer/Main/UI/Pages/Home/ProductsTab.cs [hunk 6]
```diff
@@ -13,6 +13,7 @@ class ProductsTab : BasePage
 
         public SenderForm GetSenderForm()
         {
+            CommonActions.WaitUntil(2);
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Home/ProductsTab.cs]: Why we are using this wait ? Is it not possible to use WaitUntilElementIsDisplayed(IWebElement webElement)??

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/NoPrintingOfReceiptPrinter.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿namespace IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/NoPrintingOfReceiptPrinter.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/NoPrintingOfReceiptPrinter.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+{
+    using IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer.OrderOption;
+    using IDT.MoneyTransfer.Main.Utilities;
+    using OpenQA.Selenium;
+
+    public class NoPrintingOfReceiptPrinter : IPrinterStrategy
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/NoPrintingOfReceiptPrinter.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/IPrinterStrategy.cs [hunk 9]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/IPrinterStrategy.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/IPrinterStrategy.cs [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+using IDT.MoneyTransfer.Main.Utilities;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/IPrinterStrategy.cs]: Fix according to Code standards

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/IPrinterStrategy.cs [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+using OpenQA.Selenium;
+
+namespace IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer.OrderOption
+{
+    public interface IPrinterStrategy
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/IPrinterStrategy.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/PrinterService.cs [hunk 12]
```diff
@@ -0,0 +1,12 @@
+﻿using OpenQA.Selenium;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/PrinterService.cs]: Fix according to Code Standards 
Add Header Doc

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/PrinterService.cs [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer.OrderOption
+{
+    public class PrinterService
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/PrinterService.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/PrinterService.cs [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+    {
+        public void PrinterMethod(IPrinterStrategy iPrinterStrategy, IWebElement iWebElement)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/OrderOption/PrinterService.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/ReceiptPage.cs [hunk 15]
```diff
@@ -28,6 +28,7 @@ public string GetNotificationMessage()
             IWebElement textNofication = notificationModal.FindElement(By.CssSelector("[class='ui-pnotify-text']"));
             string successMessage = textNofication.FindElement(By.TagName("b")).Text;
 
+            CommonActions.WaitUntil(3);
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/ReceiptPage.cs]: We need to check this. It should be used if we don't have more options

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/RegularPrinter.cs [hunk 16]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/RegularPrinter.cs]: Fix according to Code Standards and Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/SendingAmountForm.cs [hunk 17]
```diff
@@ -16,12 +16,15 @@ class SendingAmountForm: BasePage
         [IDTFindBy(File = "SendingAmountFormLocators", Locator = "BtnSave")]
         private IWebElement BtnSave;
 
-        public DeliverMethodForm DeliverMethod(int moneyAmount, string recievingCurrency)
+        public DeliverMethodForm DeliverMethod(double moneyAmount, string recievingCurrency)
         {
+            VariablesGlobals.MoneyAmount = moneyAmount;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/SendingAmountForm.cs]: I think that it should be GlobalVariables

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/SmallPrinter.cs [hunk 18]
```diff
@@ -0,0 +1,12 @@
+﻿namespace IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/SmallPrinter.cs]: Fix all class with Code standards and Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/SmallPrinter.cs [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+{
+    using System;
+    using IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer.OrderOption;
+    using IDT.MoneyTransfer.Main.Utilities;
+    using OpenQA.Selenium;
+
+    internal class SmallPrinter : IPrinterStrategy
+    {
+        public void Printer(IWebElement iWebElement) => CommonActions.RadioButtonCheck(iWebElement);
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/SmallPrinter.cs]: Is this a short form ?

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 20]
```diff
@@ -1,11 +1,16 @@
 ﻿namespace IDT.MoneyTransfer.Main.UI.Pages.MoneyTransfer
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add Header documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 21]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+                    new PrinterService().PrinterMethod(new RegularPrinter(), RdoPrintRegular);
+                    break;
+                case Constants.Small:
+                    new PrinterService().PrinterMethod(new SmallPrinter(), RdoPrintSmall);
+                    break;
+                case Constants.NoPrinter:
+                    new PrinterService().PrinterMethod(new NoPrintingOfReceiptPrinter(), RdoNoPrint);
+                    break;
+                default:
+                    throw new Exception("Unknown carrier");
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Update message according to Print option

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 22]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+        private IWebElement TxtTotal;
+
+        [IDTFindBy(File = "TransferAmountFormLocators", Locator = "BtnCalculate")]
+        private IWebElement BtnCalculate;
+
+        [IDTFindBy(File = "TransferAmountFormLocators", Locator = "TxtReceiving")]
+        private IWebElement TxtReceiving;
+
+
+        public void FillTransferAmountForm(string printingOption)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 23]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+                    throw new Exception("Unknown carrier");
+
             }
-            else if (printingOption.Equals("Small"))
+
+            VariablesGlobals.AmountToReceive = GetTxtReceiving();
+
+        }
+
+        public ReceiptPage ClickBtnProcess()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 24]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+        {
+            CommonActions.ClickWebElement(this.BtnProcess);
+            return new ReceiptPage();
+        }
+
+        public string GetTxtReceiving()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 25]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+        {
+            return this.TxtReceiving.GetAttribute("value");
+        }
+
+        public double GetTxtSellRate()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 26]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+
+                Console.WriteLine("It can't convert to double", e.Message);
             }
 
-            CommonActions.ClickWebElement(this.BtnProcess);
-            return new ReceiptPage();
+            return resulSellRate;
         }
 
+        public void FillTranferEntity()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add Documentation

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 27]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+            string[] rate = this.TxtSellRate.Text.Split(' ');
+            try
             {
+                resulSellRate = double.Parse(rate[3]);
             }
-            else if (printingOption.Equals("No printing of receipt"))
+            catch (Exception e)
             {
+
+                Console.WriteLine("It can't convert to double", e.Message);
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: If we don't want to that the test continue we should raise an exception and not a Console.WriteLine

File: IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs [hunk 28]
```diff
@@ -22,24 +27,81 @@ public class TransferAmountForm: BasePage
...
+            return resulSellRate;
         }
 
+        public void FillTranferEntity()
+        {
+            VariablesGlobals.TrasferDetailsEntity = new TransferDetailsEntity { MoneyAmount = VariablesGlobals.MoneyAmount, SellRate = this.GetTxtSellRate(), Receive = this.GetTxtReceiving() };
+        }
     }
-}
+}
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/MoneyTransfer/TransferAmountForm.cs]: Add empty line to the end of file

File: IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs [hunk 29]
```diff
@@ -0,0 +1,28 @@
+﻿namespace IDT.MoneyTransfer.Main.UI.Pages.Notify
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs]: Add Header Doc

File: IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs [hunk 30]
```diff
@@ -0,0 +1,28 @@
...
+{
+    using System;
+    using IDT.Core.Automation.Page.Base;
+    using IDT.Core.Automation.Selenium;
+    using IDT.Core.Automation.WebDriver;
+    using IDT.MoneyTransfer.Main.UI.Pages.Reports;
+    using OpenQA.Selenium;
+
+    internal class NotifyModals : BasePage
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs]: - Add documentation
- Is this a descriptive name ? 
- Do you think that this class can be used for all the projects ? if the answer is yes move to core

File: IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        public string GetMessageNotify()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs [hunk 32]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            IWebElement msgNotify = WebDriverManager.Instance.GetWebDriver.FindElement(By.CssSelector(".ui-pnotify-text"));
+            IJavaScriptExecutor executor = (IJavaScriptExecutor)WebDriverManager.Instance.GetWebDriver;
+            executor.ExecuteScript("arguments[0].value", msgNotify);
+            return msgNotify.Text;
+        }
+
+        public ReportTransaction ClickLinkNotify()
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Notify/NotifyModals.cs]: Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs [hunk 33]
```diff
@@ -0,0 +1,37 @@
+﻿namespace IDT.MoneyTransfer.Main.UI.Pages.Reports
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs]: Add Header doc

File: IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs [hunk 34]
```diff
@@ -0,0 +1,37 @@
...
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using IDT.Core.Automation.Page.Base;
+    using IDT.Core.Automation.WebDriver;
+    using OpenQA.Selenium;
+
+    public class ReportTransaction : BasePage
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs]: Add doc

File: IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs [hunk 35]
```diff
@@ -0,0 +1,37 @@
...
+    {
+       public string GetTxtAmountToReceive(string data)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs]: Add doc

File: IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs [hunk 36]
```diff
@@ -0,0 +1,37 @@
...
+       {
+            string parentWindowHandle = WebDriverManager.Instance.GetCurrentWindow();
+            string resultAmountToReceive = string.Empty;
+            try
+            {
+                WebDriverManager.Instance.GetWebDriver.SwitchTo().Window(WebDriverManager.Instance.GetWebDriver.WindowHandles.Last());
+                IList<IWebElement> Lists = WebDriverManager.Instance.GetWebDriver.FindElements(By.CssSelector("div[class='col-md-8 col-xs-7 control-label-amount text-right']"));
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Reports/ReportTransaction.cs]: Why we don't have this locator defined at the beginning of the class?

File: IDT.MoneyTransfer/Main/Utilities/CommonActions.cs [hunk 37]
```diff
@@ -68,5 +68,10 @@ public static void LogoutPortal()
             WaitUntilElementIsVisible("a span[class='hidden-xxs']");
             WebDriverManager.Instance.GetWebDriver.FindElement(By.CssSelector("a span[class='hidden-xxs']")).Click();
         }
+
+        public static void RadioButtonCheck(IWebElement iWebElement)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/CommonActions.cs]: Add doc... and I think that it should be moved to the core

File: IDT.MoneyTransfer/Main/Utilities/Constants.cs [hunk 38]
```diff
@@ -9,5 +9,8 @@ public class Constants
         public const string TxtSearchState = "input[class='select2-search__field']";
         public const string PhoneNumberExistentSender = "5555555555";
         public const string LastNameExistentSender = "Vayne";
+        public const string Regular = "Regular";
+        public const string Small = "Small";
+        public const string NoPrinter = "No printing of receipt";
     }
 }
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Constants.cs]: add empty line to the end of file

File: IDT.MoneyTransfer/Main/Utilities/EnumUtilities/CountryMoney.cs [hunk 39]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/EnumUtilities/CountryMoney.cs]: Update all class with code standards and doc

File: IDT.MoneyTransfer/Main/Utilities/Utils.cs [hunk 40]
```diff
@@ -4,6 +4,8 @@
     using System.Collections.ObjectModel;
     using System.Threading;
     using IDT.Core.Automation.WebDriver;
+    using IDT.MoneyTransfer.Main.UI.Entities;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Utils.cs]: Add Header doc

File: IDT.MoneyTransfer/Main/Utilities/Utils.cs [hunk 41]
```diff
@@ -4,6 +4,8 @@
...
+    using System.Collections.Generic;
 
     public class Utils
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Utils.cs]: add doc

File: IDT.MoneyTransfer/Main/Utilities/Utils.cs [hunk 42]
```diff
@@ -16,5 +18,22 @@ public static ReadOnlyCollection<string> GetCurrentWindows()
         {
              return WebDriverManager.Instance.GetCurrentWindows();
         }
+
+        public static string GetRoudingRules(CountryMoney country)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Utils.cs]: add doc

File: IDT.MoneyTransfer/Main/Utilities/Utils.cs [hunk 43]
```diff
@@ -16,5 +18,22 @@ public static ReadOnlyCollection<string> GetCurrentWindows()
...
+        {
+            Func<TransferDetailsEntity, string> receiveroutingMxn = ReceiveRoutingMxn;
+
+            // Func<TransferDetailsEntity, string> receiveroutingCop = s => s.Age > 12 && s.Age < 20;
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Utils.cs]: remove commented code

File: IDT.MoneyTransfer/Main/Utilities/Utils.cs [hunk 44]
```diff
@@ -16,5 +18,22 @@ public static ReadOnlyCollection<string> GetCurrentWindows()
...
+
+            Dictionary<CountryMoney, string> strategyCountryMoney = new Dictionary<CountryMoney, string>();
+            strategyCountryMoney.Add(CountryMoney.MXN, receiveroutingMxn(VariablesGlobals.TrasferDetailsEntity));
+            // strategyCountryMoney.Add(CountryMoney.COP, receiveroutingCop(VariablesGlobals.TrasferDetailsEntity));
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Utils.cs]: remove commented code

File: IDT.MoneyTransfer/Main/Utilities/Utils.cs [hunk 45]
```diff
@@ -16,5 +18,22 @@ public static ReadOnlyCollection<string> GetCurrentWindows()
...
+            return strategyCountryMoney[country];
+        }
+
+        public static string ReceiveRoutingMxn(TransferDetailsEntity transferDetailsEntity)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/Utils.cs]: Is it necessary to do this method as static ? is it used in other classes ?

File: IDT.MoneyTransfer/Main/Utilities/VariablesGlobals.cs [hunk 46]
```diff
@@ -0,0 +1,30 @@
+﻿#pragma warning disable SA1633 // File must have header
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/VariablesGlobals.cs]: - remove pragma warnings 
- Add header doc 
- Fix class according to Code standards and Documentation

File: IDT.MoneyTransfer/Main/Utilities/VariablesGlobals.cs [hunk 47]
```diff
@@ -0,0 +1,30 @@
...
+using IDT.MoneyTransfer.Main.UI.Entities;
+
+namespace IDT.MoneyTransfer.Main.Utilities
+{
+    internal class VariablesGlobals
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/VariablesGlobals.cs]: I think that the correct name should be GlobalVariables

File: IDT.MoneyTransfer/Main/Utilities/VariablesGlobals.cs [hunk 48]
```diff
@@ -0,0 +1,30 @@
...
+
+        public static string MxpectedResult { get; set; }
+
+        public static string ReceivingCurrency { get; set; }
+
+        public static string AmountToReceive { get; set; }
+
+        public static TransferDetailsEntity TrasferDetailsEntity { get; set; }
+
+        public static void GetTimeWait(string precondition, int time)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/Utilities/VariablesGlobals.cs]: - What exactly does this ? Do we only wait for the word After ? 
- Add Doc

File: IDT.MoneyTransfer/Tests/UI/Features/UI/MoneyTransfer.feature [hunk 49]
```diff
@@ -13,6 +13,7 @@ Scenario: Verify that a new sender is able to send money to new recipient
 		And I send "10" USD with "MXN" Receiving Currency
 		And I select "Cash Pickup" on Delivery Method
 		And I select "Regular" for Printing options
+		And I press the "process" Button
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/Features/UI/MoneyTransfer.feature]: I think that it should be: I press "Process" button

File: IDT.MoneyTransfer/Tests/UI/Features/UI/TransferAmount.feature [hunk 50]
```diff
@@ -0,0 +1,19 @@
...
+		And I fill the new Recipient form with following data:
+			| PhoneNumber			|FirstName		|  LastName     | MaindenName	| Email							| Addres1	|Addres2	|PostalCode     |City						|State					|
+			|6666666666				| recipient03	| vayne         | rivera		|auto_recipient03@mailinator.com|av test	|av test	|78975			| DISTRITO FEDERAL, MX-BCN  |BAJA CALIFORNIA NORTE	|
+		And I send "10" USD with "MXN" Receiving Currency
+		And I select "Cash Pickup" on Delivery Method
+		And I calculate the transfer detaill to the rounding rules from "MXN"
+		And I select "Regular" for Printing options
+		And I press the "process" Button 
+	Then The receipt should be displayed
+	    And The receipt should be displayed with the correct according to the roundng rules
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/Features/UI/TransferAmount.feature]: Add empty line to the end of file

File: IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs [hunk 51]
```diff
@@ -1,5 +1,6 @@
 ﻿namespace IDT.MoneyTransfer.Tests.UI.StepsDefinitions
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs]: Add Header doc

File: IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs [hunk 52]
```diff
@@ -8,6 +9,8 @@
     using NUnit.Framework;
     using TechTalk.SpecFlow;
     using TechTalk.SpecFlow.Assist;
+    using IDT.MoneyTransfer.Main.UI.Pages.Notify;
+    using IDT.MoneyTransfer.Main.UI.Pages.Reports;
 
     [Binding]
     public class MoneyTransferSteps
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs]: Add doc

File: IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs [hunk 53]
```diff
@@ -98,12 +101,38 @@ public void WhenISelectOnDeliveryMethod(string deliverMethod)
...
+        [When(@"I press the ""(.*)"" Button")]
+        public void WhenIPressTheButton(string p0)
+        {
+            this.receiptPage = this.transferAmount.ClickBtnProcess();
         }
 
+        [When(@"I calculate the transfer detaill to the rounding rules from ""(.*)""")]
+        public void WhenICalculateTheTransferDetaillToTheRoundingRulesFrom(string country)
+        {
+
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs]: remove this empty line

File: IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs [hunk 54]
```diff
@@ -98,12 +101,38 @@ public void WhenISelectOnDeliveryMethod(string deliverMethod)
...
+            this.transferAmount.FillTranferEntity();
+            switch (country)
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs]: All these values should be get from UI and compare with our constants

File: IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs [hunk 55]
```diff
@@ -121,6 +150,18 @@ public void ThenTheNotificationMessageShouldBe(string expectedMessage)
             CommonActions.LogoutPortal();
         }
 
+        [Then(@"The receipt should be displayed with the correct according to the roundng rules")]
+        public void ThenTheReceiptShouldBeDisplayedWithTheCorrectAccordingToTheRoundngRules()
+        {
+            this.receiptPage.GetNotificationMessage();
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs]: Why is need this ? if it is required we should have a specific step for this

File: IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs [hunk 56]
```diff
@@ -121,6 +150,18 @@ public void ThenTheNotificationMessageShouldBe(string expectedMessage)
...
+            ReportTransaction reportMoney = new NotifyModals().ClickLinkNotify();
+            string actullyResult = VariablesGlobals.MxpectedResult + " " + VariablesGlobals.ReceivingCurrency;
+            string expectedResult = reportMoney.GetTxtAmountToReceive(actullyResult);
+            Assert.AreEqual(expectedResult, actullyResult);
+            CommonActions.LogoutPortal();
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Tests/UI/StepsDefinitions/MoneyTransferSteps.cs]: Why we have this line here ?

File: IDT.MoneyTransfer/Main/UI/Entities/TransferDetailsEntity.cs [hunk 57]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IDT.MoneyTransfer.Main.UI.Entities
+{
+    public class TransferDetailsEntity
```
Comment (wMercadoJala) [inline IDT.MoneyTransfer/Main/UI/Entities/TransferDetailsEntity.cs]: - [ ] Add documentation

File: IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs (hunk 5 above)
Comment (rAguilarJala) [inline IDT.MoneyTransfer/Main/UI/Pages/Home/HomePage.cs]: ok, I changed the wait by WaitUntilElementIsDisplayed(IWebElement webElement)

File: IDT.MoneyTransfer/Main/UI/Pages/Home/ProductsTab.cs (hunk 6 above)
Comment (rAguilarJala) [reply IDT.MoneyTransfer/Main/UI/Pages/Home/ProductsTab.cs]: The same I changed the wait.

**Comment by rAguilarJala** [comment]
run-2017-10-11_14-10-03

**Pull request merged**